!feat/async and !feat/interrupts (breaking changes)

elpiel opened the pull request:
I was not able to make the interrupt pin trigger despite double-checking all settings, making sure register values are correct.

I've tested with 2 different boards but no interrupt pin nor interrupt register is ever raised.

- `async`/`blocking` features (`blocking` enabled by default to keep aligned with current version)
- Config & interrupt setup for BNO055 (untested, register nor pin get triggered)
- Various improvements for Debug & defmt::Format

billythedummy committed:
Add interrupts and interrupts mask

billythedummy committed:
add accelerometer interrupt settings

billythedummy committed:
add interrupt settings

billythedummy committed:
need to be in config mode to change interrupt configs, add interrupt reset

billythedummy committed:
ACC_AM, cargo fmt

billythedummy committed:
add acc config

billythedummy committed:
add set_acc_config

billythedummy committed:
make acc config fields pub

elpiel committed:
Merge remote-tracking branch 'billythedummy/master' into feat/configs-and-interupts

Signed-off-by: Lachezar Lechev <elpiel93@gmail.com>

elpiel committed:
feat: Interrupts for Accelerometer and Gyroscope
feat: add 9DOF & 6DOF methods
chore: move Accelerometer Gyroscope interrupt config structs to own file

Signed-off-by: Lachezar Lechev <elpiel93@gmail.com>

elpiel committed:
chore: move SYS_TRIGGER register values to separate struct

Signed-off-by: Lachezar Lechev <elpiel93@gmail.com>

elpiel committed:
chore: update defmt impl for status code

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
feat: async impl

elpiel committed:
feat: Acc/gyr/mag config

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
fix: FromPrimitive + conflicting default impl

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
fix: compilation

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
chore: custom defmt::Format impls

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
fix: reg. value and try to fix interrupts

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

zeenix commented:
@elpiel Hi there! This PR is so cool and useful. Thanks for doing this. Do you intend to finish it? Do you recall why the CI was failing (the log is no longer accessible as it has expired)?

elpiel committed:
fix: lints

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
fix: enable blocking feature by default

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
fix: allow clippy type_complexity for dof9 & dof9_fixex

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
fix: interrupts reg value test

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
fix(ci): install x86_64-musl

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel commented on .github/workflows/rust.yml:
```diff
@@ -67,6 +67,8 @@ jobs:
           tool: cargo-fuzz
       - name: Fuzz (nightly only)
         if: ${{ matrix.rust == 'nightly' }}
+        env:
+          RUSTFLAGS: "-C target-feature=-crt-static"
         run: |
           cargo +nightly fuzz run init -- -max_total_time=900
           cargo +nightly fuzz run calibrate -- -max_total_time=1800
```
For musl targets the `cargo fuzz` fails:

- Issue: https://github.com/rust-fuzz/cargo-fuzz/issues/397
- See workflow run: https://github.com/eupn/bno055/actions/runs/19533061607/job/55920473048

elpiel committed:
fix(ci): install musl-tools for nightly with fuzz tool

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
fix(ci): use cargo install for cargo-fuzz because of musl

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
fiix(ci): bump msrv

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
fix: features and calibrate example
chore: rename `defmt-03` feature to `defmt`
docs: README - add missing features

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
fix: cargo.toml features

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
fix(ci): remove musl target

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
fix(ci): docs build

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

elpiel committed:
fix(fuzz): bno055 dependency

Signed-off-by: Lachezar Lechev <lachezar@lechev.space>

zeenix commented:
@eupn Do you think you'd have time to review this PR and roll out a release after it's merged, in the near future? :)

zeenix commented:
> @eupn Do you think you'd have time to review this PR and roll out a release after it's merged, in the near future? :)

I guess not. :( @elpiel I wonder if this is now an unmaintained crate, given that the last commit in the repo was from a year ago and some of your PRs haven't gotten any comments for more than half a year now.
